pkg/controller/nodenetworkstate/nodenetworkstate_controller.go

```diff
@@ -38,6 +38,10 @@ func newReconciler(mgr manager.Manager) reconcile.Reconciler {
 	return &ReconcileNodeNetworkState{client: mgr.GetClient(), scheme: mgr.GetScheme()}
 }
 
+func desiredState(object runtime.Object) nmstatev1.State {
+	return object.(*nmstatev1.NodeNetworkState).Spec.DesiredState
```

Comment:
We should make this bit safer. It is code in controller that might panic if object is not the expected type (unlikely, but trust nobody, maybe it's fine now, but we might forget, use the function in another riskier context and fail hard). Would you please try to cast the type using case and return error in case it doesn't match?